test/unit/steps/do-query/do-get.js

```diff
@@ -0,0 +1,108 @@
+import {expect} from "chai";
+import ResourceTypeRegistry from "../../../../src/ResourceTypeRegistry";
+import doGet from "../../../../src/steps/do-query/do-get";
+
+class TestAdapter {
+  static getReferencedType(model, path) {
+    if (path === "some-relationship") {
+      return "dasherize-off-type";
+    }
+    return "dasherize-on-type";
+  }
+  find(type, idOrIds, fields, sorts, filters, includes) {
+    return {
+      then() {
+        return { type, idOrIds, fields, sorts, filters, includes };
+      }
+    };
+  }
+}
+
+describe("GET requests", () => {
+
+  let registry;
+  let dbAdapter = new TestAdapter();
+
+  before(() => {
+    registry = new ResourceTypeRegistry();
+    registry.type("dasherize-on-type", { dbAdapter: dbAdapter });
+    registry.type("dasherize-off-type", {
+      dbAdapter: dbAdapter,
+      behaviors: {
+        dasherizeOutput: { enabled: false }
+      }
+    });
+  });
+
+  describe("parsing query params", () => {
+    it("should camelize fields param according to type behavior", () => {
+      let request = {
+        type: "dasherize-on-type",
+        queryParams: {
+          fields: {
+            "dasherize-on-type": "a-dasherized-field",
+            "dasherize-off-type": "another-dasherized-field"
+          }
+        }
+      };
+
+      let passedToAdapter = doGet(request, {}, registry);
+      expect(passedToAdapter.fields).to.deep.equal({
```

Comment:
Oh, actually I remember this now. It is a little weird, but it does exactly what I meant it to do. The intention was to inspect what arguments are passed to the dbAdapter, and no more. So my `TestAdapter`’s `find` method is not intended to mock the actual behavior of a dbAdapter’s `find`. It’s simply meant to spit back out the arguments that it was passed so that we can ensure that the do-get step is working. Leaving this as-is. :white_check_mark: